internal/core/crdt/composite.go

```diff
@@ -41,13 +36,23 @@ type CompositeDAGDelta struct {
 	// Status represents the status of the document. By default it is `Active`.
 	// Alternatively, if can be set to `Deleted`.
 	Status client.DocumentStatus
-	// SubDAGS should not be marshalled as they are already
-	// stored as links in the DAG blocks. They are needed here to
-	// hold on to them for the block creation.
-	SubDAGs []core.DAGLink `json:"-"`
 }
 
-var _ core.CompositeDelta = (*CompositeDAGDelta)(nil)
+var _ core.Delta = (*CompositeDAGDelta)(nil)
+
+// IPLDSchemaBytes returns the IPLD schema representation for the type.
+//
+// This needs to match the [CompositeDAGDelta] struct or [coreblock.mustSetSchema] will panic on init.
+func (delta *CompositeDAGDelta) IPLDSchemaBytes() []byte {
+	return []byte(`
+	type CompositeDAGDelta struct {
```

Comment:
Since it's a small amount of code/text, I find it more convenient to have it right there as you read the Go code. I don't see the benefit of separating them into their own files.

Reply:
If we were to split them into their own files we'd probably want to force a one-one file correspondence with the Go code too otherwise which `.proto` file goes with which `.go` file would probably be a bit confusing - and I'm not sure we'd want to force that on the Go code, and splitting the go files would be better done in another PR anyway (I've not checked to be sure, but I don't think they are one type per file atm)

Reply:
They are one type per file :)

Reply:
Its just A) Somewhat ugly B) not well structured, and doesn't lend itself to working with all the IPLD schemas independent of the delta implementations

Reply:
A) That is subjective. 
B.1) It's well structured in the sense that it is right next to the Go type that represents it. 
B.2) Why would we want to use them independently of the delta structs? They are merely a representation of the same thing. The IPLD schema is used to help translate from an IPLD node representation to the corresponding Go struct.

Reply:
A) absolutely 
B1) I meant not well structured as it's just some arbitrary inline byte string inside a method
B2) Now that we have dedicated IPLD schemas defined, it opens up a whole world of things we can do with them. Eg generate non Go serialization bindings, use the schemas in docs,  can have util functions that operate on the schemas themselves, etc.

The reason I made the proto analogy, is that much like protobuf it actually has nothing to do with Go. IPLD schemas are general and language independent, and we're losing out on some potential functionally to have them inline like this.

This isn't a PR blocker, just a general design point for this new concept being brought into the codebase with this PR